lib/clAdapters/googleAdapter.js

```diff
@@ -1,37 +1,386 @@
+      var messageData = JSON.parse(messageJson.replace(new RegExp('\\"', 'g'),'"'));
+      //console.log('-----');
+      //console.log(messageData);
+      return Promise.reject(messageData);
+    } else {
+      return Promise.reject(err);
+    }
+  });
+};
+
+var getHeaderValue = function(message, headerName) {
+  var headerValues = _(message.payload.headers)
+          .filter(function(header) { return header.name === headerName; })
+          .pluck('value')
+          .value();
+  if(headerValues.length > 0) {
+    return headerValues[0];
+  } else {
+    return null;
+  }
+};
+
+var getEmailAddressObjectFromString = function(value) {
+  var returnObject = {
+    name: value,
+    address: value
+  };
+
+  if(value && value.indexOf('>') > 0) {
+    var valueArray = value.split(' ');
+    returnObject.address = valueArray[valueArray.length-1].replace('<', '').replace('>', '');
+    returnObject.name = value.replace(' ' + valueArray[valueArray.length-1], '');
+  }
+
+  return returnObject;
+}
+
+var convertEmailListToArrayOfEmailAddressObjects = function(emailList) {
+  var emailAddressObjectArray = [];
+  if(emailList) {
+    var emailArray = emailList.split(',');
+    for(var emailIter = 0; emailIter < emailArray.length; emailIter++) {
+      emailAddressObjectArray.push(getEmailAddressObjectFromString(emailArray[emailIter]));
+    }
+  }
+
+  return emailAddressObjectArray;
+}
+
+var hasLabel = function(message, labelValue) {
+  return message.labelIds.indexOf(labelValue) >= 0;
+}
+
+var mapEmailData = function(emailData) {  
+  var mappedData = [];
+
+  for (var userIter = 0; userIter < emailData.length; userIter++) {
+    var mappedUser = {
+      email: emailData[userIter].email,
+      filterStartDate: emailData[userIter].filterStartDate,
+      filterEndDate: emailData[userIter].filterEndDate,
+      data: [],
+      success: emailData[userIter].success,
+      errorMessage: emailData[userIter].errorMessage
+    };
+
+    if(emailData[userIter].success) {
+      for (var i = 0; i < emailData[userIter].data['messages'].length; i++) {
+        var originalEmailMessage = emailData[userIter].data['messages'][i];
+        var mappedEmailMessage = {};
+
+        mappedEmailMessage = originalEmailMessage;
+        var messageData = originalEmailMessage.messageData;
+
+        mappedEmailMessage.messageId = originalEmailMessage.messageId;
+        mappedEmailMessage.conversationId = messageData.threadId;
+        mappedEmailMessage.dateTimeSent = moment(new Date(getHeaderValue(messageData, 'Date'))).utc().toISOString();
+
+        var dateReceived = getHeaderValue(messageData, 'Received');
+        if(dateReceived) {
+          var datePartOfValue = dateReceived.split(';')[1];
+          mappedEmailMessage.dateTimeReceived = moment(new Date(datePartOfValue)).utc().toISOString();
+        }
+
+        mappedEmailMessage.importance = "Normal";
+        if(hasLabel(messageData, 'IMPORTANT')) {
```

Comment:
Are there any potential i18n issues if the email user is in  different locale? Still non-blocking since I don't think we expect non-English, but should log as potential future item